src/Server/Stream.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @constructor
+ * @param {Element}
+ */
+function Stream (element) {
+	this.stack = [element]
+	Readable.call(this)
+}
+/**
+ * @type {Object}
+ */
+Stream.prototype = Object.create(Readable.prototype, {
+	/**
+	 * @return {void}
+	 */
+	_read: {value: function read () {
+		this.push(this.stack.length ? toChunk(this.stack.pop(), this.stack) : null)
+	}}
+})
+
+/**
+ * @param {function=}
+ * @return {Stream}
+ */
+function toStream (callback) {
+	var readable = new Stream(this)
+
+	if (typeof callback === 'function')
+		readable.on('end', callback)
+
+	return readable
+}
+
+/**
+ * @param {Element} element
+ * @param {Array} stack
+ * @return {string}
+ */
+function toChunk (element, stack) {
+	while (element.flag === ElementComponent)
+		element = (componentMount(element), element.children)		
+
+	var type = element.type
+	var children = element.children
+	var length = children.length
+	var output = ''
+
+	switch (element.flag) {
+		case ElementPromise:
+			return void element.type.then(function (element) {
+					toChunk(commitElement(element), stack)
+			})
+		case ElementText:
+			output = escapeText(children)
+			break
+		case ElementNode:
+			output = '<' + type + toProps(element, element.props) + '>'
+				
+			if (element.html) {
+				output += element.html
+				element.html = ''
```

Comment:
does this mean support for circular node structure? why reset it?

Reply:
This ensures that hoisted elements are reset to an original state.

```js
const element = <div>...<div>
dio.renderToStream(element)
// element.html would not be falsy if we did not reset it.
```